client/my-sites/site-settings/disconnect-site/why/index.jsx

```diff
@@ -0,0 +1,60 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import page from 'page';
+
+/**
+ * Internal dependencies
+ */
+import DocumentHead from 'components/data/document-head';
+import Main from 'components/main';
+import DisconnectReasonsMenu from './disconnect-reasons-menu';
+import { getSelectedSiteId, getSelectedSiteSlug } from 'state/ui/selectors';
+import { isJetpackSite } from 'state/sites/selectors';
+import FormattedHeader from 'components/formatted-header';
+
+class DisconnectSite extends Component {
+	componentDidMount() {
+		this.verifySiteIsJetpack();
+	}
+
+	componentDidUpdate() {
+		this.verifySiteIsJetpack();
+	}
+
+	verifySiteIsJetpack() {
```

Comment:
I'm noticing the non-JP site redirection code has been borrowed from `ManageConnection`. Can we abstract it in a HoC component that we can reuse, so we won't have to repeat all of it multiple times (I noticed that you're also using it more than once)?